code/ship/shiphit.cpp

```diff
@@ -1000,6 +1003,17 @@ float do_subobj_hit_stuff(object *ship_objp, const object *other_obj, const vec3
 				damage_to_apply *= ss_dif_scale;
 			}
 
+			if (j == 0) {
+				auto subsys_impact_real = ConditionData {
+					ImpactCondition(subsystem->armor_type_idx),
+					HitType::SUBSYS,
+					damage_to_apply,
+					subsystem->current_hits,
+					subsystem->max_hits,
+				};
+				subsys_impact = &subsys_impact_real;
+			}
+
```

Comment:
This will not work.
`subsys_impact_real` will be deallocated once this if scope closes, and the pointer will point to garbled memory.
You've basically got three options of varying levels of annoyingness:
1. (likely preferred) pass out a ConditionData object, and not a ConditionData pointer.
2. pass out a std::unique_ptr<ConditionData> object allocated with std::make_unique
3. Allocate a ConditionData Object in all callers of this function, and pass a pointer to that into this function, then fill the individual fields of the object passed in.